src/twisted/python/htmlizer.py

```diff
@@ -6,23 +6,33 @@
 HTML rendering of Python source.
 """
 
-import tokenize, cgi, keyword
+from twisted.python.compat import _PY3
+
+import tokenize, keyword
+if _PY3:
+    from html import escape
+else:
+    from cgi import escape
+
 from . import reflect
 
 class TokenPrinter:
 
     currentCol, currentLine = 0, 1
     lastIdentifier = parameters = 0
+    encoding = "utf-8"
 
     def __init__(self, writer):
         self.writer = writer
 
+
     def printtoken(self, type, token, sCoordinates, eCoordinates, line):
+        if _PY3 and type == tokenize.ENCODING:
```

Comment:
The second line of defense is module-level imports and `ImportError`.  This should instead be something like:

```python
try:
    from tokenize import ENCODING
except ImportError:
    ENCODING = 'ENCODING_NOT_DEFINED_HERE'

This way, if a dot-release of py2 ever starts defining encoding (which it might do, given that `-*- coding -*-` comments still do something on py2) it will silently start working, rather than forcibly breaking on py2 even if it might work.
```